cfme/tests/services/test_rest_services.py

```diff
@@ -116,121 +137,156 @@ def test_edit_multiple_services(self, rest_api, services):
+    def test_delete_service_post(self, rest_api, services):
+        """Tests deleting services from detail using POST method.
+
+        Metadata:
+            test_flag: rest
+        """
+        for service in services:
+            service.action.delete(force_method="post")
+            assert rest_api.response.status_code == 200
+            with error.expected("ActiveRecord::RecordNotFound"):
+                service.action.delete(force_method="post")
+            assert rest_api.response.status_code == 404
+
+    def test_delete_service_delete(self, rest_api, services):
+        """Tests deleting services from detail using DELETE method.
+
+        Metadata:
+            test_flag: rest
+        """
+        for service in services:
+            service.action.delete(force_method="delete")
+            assert rest_api.response.status_code == 204
+            with error.expected("ActiveRecord::RecordNotFound"):
+                service.action.delete(force_method="delete")
+            assert rest_api.response.status_code == 404
 
     def test_delete_services(self, rest_api, services):
+        """Tests deleting services from collection.
+
+        Metadata:
+            test_flag: rest
+        """
         rest_api.collections.services.action.delete(*services)
+        assert rest_api.response.status_code == 200
         with error.expected("ActiveRecord::RecordNotFound"):
             rest_api.collections.services.action.delete(*services)
+        assert rest_api.response.status_code == 404
+
+    @pytest.mark.parametrize(
+        "from_detail", [True, False],
+        ids=["from_detail", "from_collection"])
+    def test_retire_service_now(self, rest_api, services, from_detail):
+        """Test retiring a service now.
 
-    def test_retire_service_now(self, rest_api, services):
-        """Test retiring a service
-        Prerequisities:
-            * An appliance with ``/api`` available.
-        Steps:
-            * Retrieve list of entities using GET /api/services , pick the first one
-            * POST /api/service/<id> (method ``retire``)
         Metadata:
             test_flag: rest
         """
-        assert "retire" in rest_api.collections.services.action.all
-        retire_service = services[0]
-        retire_service.action.retire()
-        wait_for(
-            lambda: not rest_api.collections.services.find_by(name=retire_service.name),
-            num_sec=600,
-            delay=10,
-        )
+        if from_detail:
+            for service in services:
+                service.action.retire()
+                assert rest_api.response.status_code == 200
+        else:
+            rest_api.collections.services.action.retire(*services)
+            assert rest_api.response.status_code == 200
+
+        for service in services:
+            wait_for(
+                lambda: not rest_api.collections.services.find_by(name=service.name),
```

Comment:
**Optional**
Consider using func_args here and dropping the lambda, it will wait while find_by() returns a Trueish value.